pallets/auctions/src/lib.rs

```diff
@@ -136,195 +187,415 @@ pub mod pallet {
-		<AuctionOwnerById<T>>::insert(auction_id, &info.owner);
-		<AuctionEndTime<T>>::insert(info.end, auction_id, ());
+		<Auctions<T>>::insert(auction_id, auction.clone());
+		<AuctionOwnerById<T>>::insert(auction_id, &sender);
 
-		pallet_uniques::Pallet::<T>::freeze(RawOrigin::Signed(info.owner.clone()).into(), info.token.0, info.token.1)?;
+		pallet_uniques::Pallet::<T>::freeze(
+			RawOrigin::Signed(sender.clone()).into(),
+			general_data.token.0,
+			general_data.token.1,
+		)?;
 
-		Ok(auction_id)
-	}
+		Self::deposit_event(Event::AuctionCreated(sender, auction_id));
 
-	fn update_auction(id: Self::AuctionId, info: AuctionInfoOf<T>) -> DispatchResult {
-		<Auctions<T>>::try_mutate(id, |auction| -> DispatchResult {
-			ensure!(auction.is_some(), Error::<T>::AuctionNotExist);
-			*auction = Some(info);
-			Ok(())
-		})
+		Ok(())
 	}
 
-	fn remove_auction(id: Self::AuctionId) -> DispatchResult {
-		let auction = <Auctions<T>>::take(id).ok_or(Error::<T>::AuctionNotExist)?;
+	/// 
+	/// Validates certain aspects relevant to the update action
+	/// 
+	fn validate_update(sender: <T>::AccountId, general_data: &GeneralAuctionData<T>) -> DispatchResult {
+		ensure!(general_data.owner == sender, Error::<T>::NotAuctionOwner);
+
 		let current_block_number = frame_system::Pallet::<T>::block_number();
-		ensure!(current_block_number < auction.start, Error::<T>::AuctionAlreadyStarted);
-		pallet_uniques::Pallet::<T>::thaw(
-			RawOrigin::Signed(auction.owner.clone()).into(),
-			auction.token.0,
-			auction.token.1,
-		)?;
-		<AuctionOwnerById<T>>::remove(id);
-		<Auctions<T>>::remove(id);
+		ensure!(
+			current_block_number < general_data.start,
+			Error::<T>::AuctionAlreadyStarted
+		);
+
 		Ok(())
 	}
 
-	fn bid(bidder: Self::AccountId, id: Self::AuctionId, value: Self::Balance) -> DispatchResult {
-		<Auctions<T>>::try_mutate_exists(id, |auction| -> DispatchResult {
-			// Basic checks before a bid can be made
-			let mut auction = auction.as_mut().ok_or(Error::<T>::AuctionNotExist)?;
-			let block_number = <frame_system::Pallet<T>>::block_number();
-			ensure!(bidder != auction.owner, Error::<T>::BidOnOwnAuction);
-			ensure!(block_number > auction.start, Error::<T>::AuctionNotStarted);
-			ensure!(block_number < auction.end, Error::<T>::AuctionAlreadyConcluded);
-			ensure!(value >= auction.minimal_bid, Error::<T>::InvalidBidPrice);
-			if let Some(ref current_bid) = auction.last_bid {
-				ensure!(value > current_bid.1, Error::<T>::InvalidBidPrice);
-				// Unlock funds from the previous bid
-				<T as crate::Config>::Currency::remove_lock(AUCTION_LOCK_ID, &current_bid.0);
+	/// 
+	/// Handles auction update
+	/// 
+	fn handle_update(
+		sender: <T>::AccountId,
+		auction_id: T::AuctionId,
+		updated_auction: Auction<T>,
+		general_data: &GeneralAuctionData<T>,
+	) -> DispatchResult {
+		<Auctions<T>>::try_mutate(auction_id, |auction_result| -> DispatchResult {
+			if let Some(_auction) = auction_result {
```

Comment:
if you dont need the value at all.

```suggestion
			if  auction_result.is_some() {
```